app/main.py

```diff
@@ -97,10 +97,25 @@ def index_redirect_dir(request):
 
 routes.static(config.URL_PREFIX + 'favicon/', 'favicon')
 routes.static(config.URL_PREFIX, 'ui/dist/metube')
-
 app.add_routes(routes)
 
 
+# https://github.com/aio-libs/aiohttp/pull/4615 waiting for release
+# @routes.options(config.URL_PREFIX + 'add')
+async def add_cors(request):
+    return web.Response(text=serializer.encode({"status": "ok"}))
+
+app.router.add_route('OPTIONS', config.URL_PREFIX + 'add', add_cors)
+
+
+async def on_prepare(request, response):

```

Comment:
Aren't the Access-Control headers only relevant for the OPTIONS request? In which case it might be better to stick them in the add_cors() method above? I'm not sure about CORS mechanics in this regard, whether it needs to be in every response or only in the response to OPTIONS.

Reply:
No. Browser making two request:
- OPTIONS (without submiting data) to see if it's possible 
- POST (with submiting data)

Both responses should say that the request is allowed